src/polyfills/fullscreen.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * Polyfill for `Element.requestFullscreen()` method.
+ * https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullscreen
+ * @this {Element}
+ */
+ function requestFullscreenPolyfill() {
+   try {
+     const requestFs = this.webkitEnterFullscreen
+      || this.webkitRequestFullScreen
+      || this.requestFullscreen
+      || this.webkitEnterFullscreen
+      || this.msRequestFullscreen
+      || this.mozRequestFullScreen;
+     return requestFs.call(this);
+   } catch (e) {
+     // requestFullscreen not supported
```

Comment:
catching `requestFs.call` exceptions will make this work differently than native.
we should remove try-catch and just do:
```
if (requestFs) {
  return requestFs.call(this);
}
```
to allow the `requestFs.call` to just throw if it fails (e.g. not initiated by a user action) as native would do.